orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/SimpleTask.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019 Armory, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.Task;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.api.SimpleStage;
+import com.netflix.spinnaker.orca.api.SimpleStageInput;
+import com.netflix.spinnaker.orca.api.SimpleStageOutput;
+import com.netflix.spinnaker.orca.jackson.OrcaObjectMapper;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.lang.reflect.TypeVariable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.GenericTypeResolver;
+import org.springframework.core.ResolvableType;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class SimpleTask implements Task {
+  private SimpleStage simpleStage;
+
+  SimpleTask(@Nullable SimpleStage simpleStage) {
+    this.simpleStage = simpleStage;
+  }
+
+  private SimpleStageInput getStageInput(Stage stage) {
+    ObjectMapper objectMapper = OrcaObjectMapper.newInstance();
+
+    try {
+      List<Class<?>> cArg = Arrays.asList(SimpleStageInput.class);
+      Method method = simpleStage.getClass().getMethod("execute", cArg.toArray(new Class[0]));
+      Type inputType = ResolvableType.forMethodParameter(method, 0).getGeneric().getType();
+      Map<TypeVariable, Type> typeVariableMap =
+          GenericTypeResolver.getTypeVariableMap(simpleStage.getClass());
+
+      return new SimpleStageInput(
+          objectMapper.convertValue(
+              stage.getContext(), GenericTypeResolver.resolveType(inputType, typeVariableMap)));
+    } catch (NoSuchMethodException exeception) {
+      log.error("Cannot get method for " + simpleStage.getName() + ": " + exeception.getMessage());
+    }
+
+    return new SimpleStageInput(new Object());
+  }
+
+  @Nonnull
+  public TaskResult execute(@Nonnull Stage stage) {
+    ExecutionStatus status;
+    SimpleStageInput simpleStageInput = getStageInput(stage);
+    SimpleStageOutput output = simpleStage.execute(simpleStageInput);
+
+    switch (output.getStatus()) {
+      case TERMINAL:
```

Comment:
Reviewer thoughts on using `FAILED` rather than `TERMINAL`? It looks like we've already flipped to `COMPLETED` from `SUCCEEDED`?

Or alternatively we keep things consistent and use `SUCCEEDED` for both (leaning this way).

Reply:
I might be misinterpreting what you're saying, but you're proposing setting the status to `SUCCEEDED` if the output status is `TERMINAL`?

Reply:
Sorry, that was unclear.

In `orca-api` we introduced `COMPLETED` in place of `SUCCEEDED`. My suggestion would be to just re-use the same status types and avoid a conversion (ie. just do `ExecutionStatus.valueOf()`).

Reply:
@ajordens we can't use `ExecutionStatus` inside `orca-api`, that would lead to circular dependencies between `orca-core` and `orca-api`. But I can make the names of the statues be the same. Does that work?

Reply:
@ajordens never mind. I'm following you now. I was wrong on how `valueOf` works. Pushing up a change for that now. Is this more inline with what you were thinking?